selinux.go

```diff
@@ -33,7 +32,7 @@ func runLabelStdioPipes(stdioPipe [][]int, processLabel, mountLabel string) erro
 	}
 	for i := range stdioPipe {
 		pipeFdName := fmt.Sprintf("/proc/self/fd/%d", stdioPipe[i][0])
-		if err := label.Relabel(pipeFdName, pipeContext, false); err != nil {
+		if err := selinux.Chcon(pipeFdName, pipeContext, false); err != nil {
```

Comment:
`label.Relabel` ends up calling `selinux.Chcon`, but it checks if selinux is enabled.

Why do we need this change?

Reply:
@giuseppe `Relabel` calls `Chcon` but ends up hard-coding recurse flag to true https://github.com/opencontainers/selinux/blob/main/go-selinux/label/label_linux.go#L169, which is now configured to use `lsetxattr(` always instead of `setxattr` ( the change was done here https://github.com/opencontainers/selinux/pull/173 ) and for somereason `lsetxattr` always gives `operation not supported` for both rootless and root cases. I think that's because `lsetxattr` tries to label this special `symlink` `/proc/self/fd/%d` but instead it should try to label the actual file.

Failure is easy to reproduce if you see https://github.com/containers/podman/pull/13687 and https://github.com/containers/buildah/pull/3875

Reply:
thanks.

What happens now if we try to relabel a real symlink (not the magic `/proc/self/fd` one)?  Do we end up relabelling the file it points to?

Reply:
@giuseppe Yes I think now we follow the symlink via https://github.com/opencontainers/selinux/blob/main/go-selinux/selinux_linux.go#L338.